From 4ff23f7fca984270079de89abaca26e299a97b82 Mon Sep 17 00:00:00 2001
From: Andy Green <andy@openmoko.com>
Date: Fri, 25 Jul 2008 23:06:04 +0100
Subject: [PATCH] fix-glamofb-cmdqueue-timeout.patch

loglevl=9 can cause failure to init glamo-fb
problem seems to be too low timeout when text scrolling can
delay commandqueue going empty

Signed-off-by: Andy Green <andy@openmoko.com>
---
 drivers/mfd/glamo/glamo-fb.c |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/mfd/glamo/glamo-fb.c b/drivers/mfd/glamo/glamo-fb.c
index f0d7600..87c7420 100644
--- a/drivers/mfd/glamo/glamo-fb.c
+++ b/drivers/mfd/glamo/glamo-fb.c
@@ -548,12 +548,13 @@ static int glamofb_cursor(struct fb_info *info, struct fb_cursor *cursor)
 
 static inline int glamofb_cmdq_empty(struct glamofb_handle *gfb)
 {
+	/* DGCMdQempty -- 1 == command queue is empty */
 	return reg_read(gfb, GLAMO_REG_LCD_STATUS1) & (1 << 15);
 }
 
 void glamofb_cmd_mode(struct glamofb_handle *gfb, int on)
 {
-	int timeout = 2000;
+	int timeout = 20000;
 
 	dev_dbg(gfb->dev, "glamofb_cmd_mode(gfb=%p, on=%d)\n", gfb, on);
 	if (on) {
@@ -605,7 +606,7 @@ EXPORT_SYMBOL_GPL(glamofb_cmd_mode);
 
 int glamofb_cmd_write(struct glamofb_handle *gfb, u_int16_t val)
 {
-	int timeout = 2000;
+	int timeout = 20000;
 
 	dev_dbg(gfb->dev, "%s: waiting for cmdq empty\n",
 		__FUNCTION__);
-- 
1.5.6.3

